httpd-client/lib/project/patch.ts

```diff
@@ -13,27 +13,33 @@ import {
   union,
 } from "zod";
 
-export type PatchState =
-  | { status: "draft" }
-  | { status: "open" }
-  | { status: "archived" }
-  | { status: "merged" };
-
-const patchStateSchema = union([
-  strictObject({ status: literal("draft") }),
-  strictObject({ status: literal("open") }),
-  strictObject({ status: literal("archived") }),
-  strictObject({ status: literal("merged") }),
-]) satisfies ZodSchema<PatchState>;
+export interface PatchState {
+  status: "draft" | "open" | "archived" | "merged";
+  revision?: string;
+  commit?: string;
+}
```

Comment:
We should take the heartwood code as reference here
https://app.radicle.xyz/seeds/seed.radicle.xyz/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5/tree/master/radicle/src/cob/patch.rs#L599

To be able to rely on the types IMO we should keep the type and extend it like the following, and do the same with `patchStateSchema`
```ts
export type PatchState =
  | { status: "draft" }
  | { status: "open"; conflicts?: [string; string][] }
  | { status: "archived" }
  | { status: "merged"; revision: string; commit: string; };
```

Reply:
is this correct? doesn't seem to work:
```ts
const patchStateSchema = union([
  strictObject({
    status: literal("draft"),
  }),
  strictObject({
    status: literal("open"),
    conflicts: array(tuple([string(), string()])).optional(),
  }),
  strictObject({
    status: literal("archived"),
  }),
  strictObject({
    status: literal("merged"),
    revision: string().optional(),
    commit: string().optional(),
  }),
]) satisfies ZodSchema<PatchState>;
```

Reply:
ah, never mind. it's working!

Reply:
[fixed](https://github.com/radicle-dev/radicle-interface/compare/7152476b8803f8d00304d6c5e0ce9b3fed13da84..b1fbaf3818b3462ead0c80a22b6e8266646330f3)